### ISSUE TITLE
Add unit tests for reduce()

### ISSUE THREAD

**Issue by jdgarciauc3m** [arcosuc3m/grppi#96]
Use unit tests of `map()` as a model.

**Comment by jdgarciauc3m** [comment]
Poly tests may be added later.

**Comment by jdgarciauc3m** [comment]
I'm reopening this issue.

In branch latest, I get:

~~~
The following tests FAILED:
	  9 - reduce_test/*.static_single (Failed)
	 10 - reduce_test/*.static_multiple (Failed)
Errors while running CTest
~~~

**Comment by jdgarciauc3m** [comment]
Solved by PR #191
